src/App.tsx

```diff
@@ -15,36 +16,121 @@ export const goodsFromServer = [
+    if (sortType === SortType.Alphabet && !isReversed) {
+      return;
+    }
+
+    const sortedItems = [...items].sort((a, b) => a.localeCompare(b));
+
+    if (isReversed) {
+      sortedItems.reverse();
+    }
+
+    setItems(sortedItems);
+    setSortType(SortType.Alphabet);
+  }
+
+  function sortByLength() {
+    if (sortType === SortType.Length && !isReversed) {
+      return;
+    }
+
+    const sortedItems = [...items].sort((good1, good2) => {
+      if (good1.length === good2.length) {
+        return good1.localeCompare(good2);
+      }
+
+      return good1.length - good2.length;
+    });
+
+    if (isReversed) {
+      sortedItems.reverse();
+    }
+
+    setItems(sortedItems);
+    setSortType(SortType.Length);
+  }
+
+  const reset = () => {
+    setItems(initialItems);
+    setSortType(SortType.None);
+    setIsReversed(false);
+  };
+
+  const reverseGoods = () => {
+    const reversedGoods = [...items].reverse();
+
+    setItems(reversedGoods);
+    setIsReversed(!isReversed);
+
+    if (JSON.stringify(reversedGoods) === JSON.stringify(initialItems)) {
+      setIsReversed(false);
+    }
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortType !== SortType.Alphabet,
+          })}
+          onClick={sortByAlphabet}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortType !== SortType.Length,
+          })}
+          onClick={sortByLength}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={reverseGoods}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortType || isReversed) && (
+          <button
+            type="button"
+            className={classNames('button', 'is-danger', 'is-light')}
+            onClick={reset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {items.map(good => (
+          <li key={good} data-cy="Good">
+            {good}
+          </li>
+        ))}
```

Comment:
The `key` prop used in the `map` function should be unique for each element. If the list can contain duplicate goods, using the good's name as a key could lead to issues. Consider using a unique identifier such as an index or adding an ID property to each item.